File: book1_companion/psych_stats_with_python_companion_v0_2/scripts/r/ch06_paired_t.R
#!/usr/bin/env Rscript
args <- commandArgs(trailingOnly = FALSE)
script_arg <- args[grep('^--file=', args)][1]
root <- normalizePath(file.path(dirname(sub('^--file=', '', script_arg)), '..', '..'), mustWork = TRUE)
source(file.path(root, "scripts", "r", "common.R"))

d <- read.csv(file.path(root, "data", "ch06_paired.csv"))
diff <- d$anxiety_post - d$anxiety_pre
test <- t.test(d$anxiety_post, d$anxiety_pre, paired = TRUE)
write_result(root, "ch06", "Paired-samples t-test", "data/ch06_paired.csv", list(
  pre_mean = mean(d$anxiety_pre), post_mean = mean(d$anxiety_post),
  mean_difference_post_minus_pre = mean(diff), t_statistic = unname(test$statistic),
  degrees_of_freedom = unname(test$parameter), p_value = test$p.value,
  cohen_dz = mean(diff) / sd(diff)
))

File: book1_companion/psych_stats_with_python_companion_v0_2_1/scripts/python/ch06_paired_t.py
#!/usr/bin/env python3
from __future__ import annotations

import pandas as pd
from scipy import stats

from common import ROOT, rounded, write_result


def main() -> None:
    data_file = "data/ch06_paired.csv"
    data = pd.read_csv(ROOT / data_file)
    difference = data["anxiety_post"] - data["anxiety_pre"]
    test = stats.ttest_rel(data["anxiety_post"], data["anxiety_pre"])
    fields = {
        "pre_mean": rounded(data["anxiety_pre"].mean()),
        "post_mean": rounded(data["anxiety_post"].mean()),
        "mean_difference_post_minus_pre": rounded(difference.mean()),
        "t_statistic": rounded(test.statistic),
        "degrees_of_freedom": int(len(data) - 1),
        "p_value": rounded(test.pvalue),
        "cohen_dz": rounded(difference.mean() / difference.std(ddof=1)),
    }
    out = write_result("ch06", "Paired-samples t-test", data_file, fields)
    print(f"wrote {out}")


if __name__ == "__main__":
    main()
